source/addonHandler/__init__.py

```diff
@@ -181,26 +181,28 @@ def _getAvailableAddonsFromPath(path):
 		if p.endswith(DELETEDIR_SUFFIX): continue
 		addon_path = os.path.join(path, p)
 		if os.path.isdir(addon_path) and addon_path not in ('.', '..'):
-			log.debug("Loading add-on from %s", addon_path)
-			try:
-				a = Addon(addon_path)
-				name = a.manifest['name']
-				log.debug(
-					"Found add-on {name} - {a.version}."
-					" Requires API: {a.minimumNVDAVersion}."
-					" Last-tested API: {a.lastTestedNVDAVersion}".format(
-						name=name,
-						a=a
-					))
-				if a.isDisabled:
-					log.debug("Disabling add-on %s", name)
-				if not isAddonCompatible(a):
-					log.debugWarning("Add-on %s is considered incompatible", name)
-					_blockedAddons.add(a.name)
-				yield a
-			except:
-				log.error("Error loading Addon from path: %s", addon_path, exc_info=True)
-
+			if(len(os.listdir(addon_path))):
+				log.debug("Loading add-on from %s", addon_path)
+				try:
+					a = Addon(addon_path)
+					name = a.manifest['name']
+					log.debug(
+						"Found add-on {name} - {a.version}."
+						" Requires API: {a.minimumNVDAVersion}."
+						" Last-tested API: {a.lastTestedNVDAVersion}".format(
+							name=name,
+							a=a
+						))
+					if a.isDisabled:
+						log.debug("Disabling add-on %s", name)
+					if not isAddonCompatible(a):
+						log.debugWarning("Add-on %s is considered incompatible", name)
+						_blockedAddons.add(a.name)
+					yield a
+				except:
+					log.error("Error loading Addon from path: %s", addon_path, exc_info=True)
+			else:
+				log.error("Error loading Addon from path: %s, try reinstalling: %s", addon_path, name,exc_info=True)

```

Comment:
I don't think there is an exception active at this stage, I dont think the parameter `exc_info` is necessary.  As mentioned in the other comment, I think it would improve the readability if this block was moved under the if statement.

I haven't yet tried this out, is it true that reinstalling will work?

Reply:
Yes reinstalling works. What happens when you delete all the files is that the addon disappears from the addon list and you have to reinstall it.

Reply:
Thanks @Awalvie. Could you please update the "testing performed" section of the initial description of this PR.  Please see [our wiki page explaining the PR template for further information and examples on this](https://github.com/nvaccess/nvda/wiki/Github-pull-request-template-explanation-and-examples#testing-performed)

Reply:
Yes definitely.

Reply:
Any other changes you'd recommend making ?

Reply:
Nope, just waiting on the changes to the "testing performed" section of the initial description of this PR. Don't forget to read the [wiki page](https://github.com/nvaccess/nvda/wiki/Github-pull-request-template-explanation-and-examples#testing-performed).

Reply:
Really sorry, forgot about it. I'll do this ASAP

Reply:
@feerrenrut Just updated it. Let me know if any other changes need to be done. Thanks for the feedback.